quint/src/runtime/compile.ts

```diff
@@ -102,17 +102,67 @@ export function
 }
 
 /**
- * Parse a string that contains a Quint module and compile it to executable
+ * Compile Quint modules to JS runtime objects from the parsed and type-checked
+ * data structures. This is a user-facing function. In case of an error, the
+ * error messages are passed to an error handler and the function returns
+ * undefined.
+ *
+ * @param modules Quint modules in the intermediate representation
+ * @param sourceMap source map as produced by the parser
+ * @param lookupTable lookup table as produced by the parser
+ * @param types type table as produced by the type checker
+ * @param mainName the name of the module that may contain state varibles
+ * @returns the compilation context
+ */
+export function
+  compile(modules: QuintModule[],
+          sourceMap: Map<bigint, Loc>,
+          lookupTable: LookupTableByModule,
+          types: Map<bigint, TypeScheme>,
+          mainName: string): CompilationContext {
+  // Push back the main module to the end
+  const main = modules.find(m => m.name === mainName)
+  // when the main module is not found, we will report an error
+  const mainNotFound =
+    main ? [] : [{
+      explanation: `Main module ${mainName} not found`,
+      refs: [],
+    }]
+  const reorderedModules =
```

Comment:
I think this comment should be further down

```suggestion
  const main = modules.find(m => m.name === mainName)
  // when the main module is not found, we will report an error
  const mainNotFound =
    main ? [] : [{
      explanation: `Main module ${mainName} not found`,
      refs: [],
    }]
  // Push back the main module to the end
  const reorderedModules =
```

Reply:
Also, it would be nice to have an explanation why it has to be pushed to the end

Reply:
That's a good point. I have restructure the whole function body. There is no need for compiling anything if the the module is not found.